File: ios/NamazVakti/NamazVakti/Models/LocationData.swift
import Foundation

struct LocationData: Codable, Identifiable, Hashable {
    var id: UUID = UUID()
    let name: String
    let country: String
    let latitude: Double
    let longitude: Double
    let timezoneIdentifier: String
    
    // Helper to check if two locations are the same based on coordinates
    func isSameLocation(as other: LocationData) -> Bool {
        return abs(self.latitude - other.latitude) < 0.001 &&
               abs(self.longitude - other.longitude) < 0.001
    }
}

File: android/app/src/main/java/com/oktay/namaz/model/LocationData.kt
package com.oktay.namaz.model

import java.util.UUID
import kotlin.math.abs

data class LocationData(
    val id: String = UUID.randomUUID().toString(),
    val name: String,
    val country: String,
    val latitude: Double,
    val longitude: Double,
    val timezoneIdentifier: String
) {
    fun isSameLocation(other: LocationData): Boolean {
        return abs(this.latitude - other.latitude) < 0.001 &&
               abs(this.longitude - other.longitude) < 0.001
    }
}
